File: tsp-java/tsp-adapter-yaml/src/test/java/dev/tsp/yaml/YamlAdapterTest.java
package dev.tsp.yaml;

import dev.tsp.core.model.*;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

class YamlAdapterTest {

    @Test
    void yamlBasic() {
        TIRResult t = new TIRResult("greeting", "user", List.of(
            new TIRTextNode("name: "), new TIRExprNode("name", "Alice")));
        assertThat(YamlAdapter.render(t))
            .isEqualTo("# tsp template_id=greeting schema_id=user\nname: Alice");
    }

    @Test
    void yamlDoesNotEscape() {
        TIRResult t = new TIRResult("t", "s", List.of(
            new TIRExprNode("content", "<b>Hello</b>")));
        assertThat(YamlAdapter.render(t)).contains("<b>Hello</b>");
    }

    @Test
    void yamlProvenanceComment() {
        TIRResult t = new TIRResult("my-template", "my-schema", List.of(new TIRTextNode("Hello")));
        assertThat(YamlAdapter.render(t))
            .startsWith("# tsp template_id=my-template schema_id=my-schema");
    }
}

File: tsp-go/yamladapter/yaml_adapter_test.go
package yamladapter

import (
	"strings"
	"testing"

	"tsp-go/core"
)

func TestYamlBasic(t *testing.T) {
	r := core.TIRResult{TemplateID: "greeting", SchemaID: "user", Nodes: []core.TIRNode{
		{Kind: "text", Content: "name: "},
		{Kind: "expr", Field: "name", Resolved: "Alice"},
	}}
	got := Render(r)
	want := "# tsp template_id=greeting schema_id=user\nname: Alice"
	if got != want {
		t.Errorf("got %q, want %q", got, want)
	}
}

func TestYamlDoesNotEscape(t *testing.T) {
	r := core.TIRResult{TemplateID: "t", SchemaID: "s", Nodes: []core.TIRNode{
		{Kind: "expr", Field: "content", Resolved: "<b>Hello</b>"},
	}}
	got := Render(r)
	if !strings.Contains(got, "<b>Hello</b>") {
		t.Errorf("got %q", got)
	}
}

func TestYamlProvenanceComment(t *testing.T) {
	r := core.TIRResult{TemplateID: "my-template", SchemaID: "my-schema", Nodes: []core.TIRNode{
		{Kind: "text", Content: "Hello"},
	}}
	got := Render(r)
	if !strings.HasPrefix(got, "# tsp template_id=my-template schema_id=my-schema") {
		t.Errorf("got %q", got)
	}
}
